### PR TITLE
Add scripts for recommendation Infra demo

### PR DESCRIPTION
**Pull request by kusumachalasani**

### CODE REVIEW

**Comment by chandrams** [comment]
@kusumachalasani - As discussed, running the test failed with the below error:

```

URL =  http://192.168.49.2:32506/createPerformanceProfile    Response status code =  409
{"message":"Validation failed: Performance Profile already exists: resource-optimization-openshift","httpcode":409,"documentationLink":"","status":"ERROR"}

Creating the experiment...
Traceback (most recent call last):
  File "/home/csubrama/github/kusuma/autotune-demo/recommendations_Infra_demo/./recommendations_demo/recommendation_experiment.py", line 252, in <module>
    main(sys.argv[1:])
  File "/home/csubrama/github/kusuma/autotune-demo/recommendations_Infra_demo/./recommendations_demo/recommendation_experiment.py", line 249, in main
    createExpAndupdateResults(clustername,resultscsv)
  File "/home/csubrama/github/kusuma/autotune-demo/recommendations_Infra_demo/./recommendations_demo/recommendation_experiment.py", line 149, in createExpAndupdateResults
    create_expjson(clustername,"intermediate.csv")
  File "/home/csubrama/github/kusuma/autotune-demo/recommendations_Infra_demo/./recommendations_demo/recommendation_experiment.py", line 102, in create_expjson
    "CLUSTER_NAME": row["cluster_name"],
KeyError: 'cluster_name'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/csubrama/github/kusuma/autotune-demo/recommendations_Infra_demo/recommendations_demo/recommendation_validation.py", line 47, in update_recomm_csv
    with open(filename, 'r') as f:
FileNotFoundError: [Errno 2] No such file or directory: 'recommendations_data.json'
expnames_array os 

Completed
```

Command used:

```
./recommendations_demo.sh -c minikube --dataDrivenRecommendations --mode="crc" --dataDir="./recommendations_demo/crc-results/" -r
```

**Comment by kusumachalasani** [comment]
> @kusumachalasani - As discussed, running the test failed with the below error:
> 
> ```
> 
> URL =  http://192.168.49.2:32506/createPerformanceProfile    Response status code =  409
> {"message":"Validation failed: Performance Profile already exists: resource-optimization-openshift","httpcode":409,"documentationLink":"","status":"ERROR"}
> 
> Creating the experiment...
> Traceback (most recent call last):
>   File "/home/csubrama/github/kusuma/autotune-demo/recommendations_Infra_demo/./recommendations_demo/recommendation_experiment.py", line 252, in <module>
>     main(sys.argv[1:])
>   File "/home/csubrama/github/kusuma/autotune-demo/recommendations_Infra_demo/./recommendations_demo/recommendation_experiment.py", line 249, in main
>     createExpAndupdateResults(clustername,resultscsv)
>   File "/home/csubrama/github/kusuma/autotune-demo/recommendations_Infra_demo/./recommendations_demo/recommendation_experiment.py", line 149, in createExpAndupdateResults
>     create_expjson(clustername,"intermediate.csv")
>   File "/home/csubrama/github/kusuma/autotune-demo/recommendations_Infra_demo/./recommendations_demo/recommendation_experiment.py", line 102, in create_expjson
>     "CLUSTER_NAME": row["cluster_name"],
> KeyError: 'cluster_name'
> Traceback (most recent call last):
>   File "<string>", line 1, in <module>
>   File "/home/csubrama/github/kusuma/autotune-demo/recommendations_Infra_demo/recommendations_demo/recommendation_validation.py", line 47, in update_recomm_csv
>     with open(filename, 'r') as f:
> FileNotFoundError: [Errno 2] No such file or directory: 'recommendations_data.json'
> expnames_array os 
> 
> Completed
> ```
> 
> Command used:
> 
> ```
> ./recommendations_demo.sh -c minikube --dataDrivenRecommendations --mode="crc" --dataDir="./recommendations_demo/crc-results/" -r
> ```

Updated the code for the same!

**Comment by chandrams** [comment]
Above issue is resolved with the updated code.

File: recommendations_Infra_demo/ReadMe [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
```
[the remark this answers is not in the document]
Comment (chandrams) [reply recommendations_Infra_demo/ReadMe:1]: Include the usage for --validate

File: recommendations_Infra_demo/recommendations_demo.sh [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+# Copy the previous recommendationsOutput.csv and experimentOutput.csv into another for future purpose.
+if [ -e "recommendationsOutput.csv" ]; then
+	mv recommendationsOutput.csv recommendationsOutput-$(date +%Y%m%d).csv
+fi
+if [ -e "experimentOutput.csv" ]; then
+        mv experimentOutput.csv experimentOutput-$(date +%Y%m%d).csv
+fi
+
+if [ ${start_demo} -eq 1 ]; then
+	monitoring_demo_start
```
Comment (chandrams) [inline recommendations_Infra_demo/recommendations_demo.sh:426]: Script should exit with 0 (success) or 1 (failure), this is required to automate this on jenkins for --validate option

Comment (chandrams) [reply recommendations_Infra_demo/recommendations_demo.sh:434]: @kusumachalasani - Has this been done? I will need this to include the actual recommendation validation (--validate) in the jenkins job. How long does this take to run?

File: recommendations_Infra_demo/recommendations_demo/kruize/json_validate.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+import jsonschema
+from jsonschema import validate, draft7_format_checker
+
+DIRECTIONS_SUPPORTED = ("maximize", "minimize")
+
+DIRECTION_NOT_SUPPORTED = "Direction not supported!"
+JSON_NULL_VALUES = ("is not of type 'string'", "is not of type 'integer'", "is not of type 'number'")
+VALUE_MISSING = " cannot be empty or null!"
+
+exp_input_schema = {
```
Comment (dinogun) [inline recommendations_Infra_demo/recommendations_demo/kruize/json_validate.py:26]: I'd suggest maintaining this as a JSON file and we can read from it here. This way it is easier to keep track changes to the JSON

File: recommendations_Infra_demo/recommendations_demo.sh (hunk 2 above)
Comment (kusumachalasani) [reply recommendations_Infra_demo/recommendations_demo.sh:434]: Yes, exit codes are incorpoated. It takes around 30mins to complete the run to validate 4 csv files (which has 15 days data) as the recommendations are generated for every timestamp.

File: remote_monitoring_demo/recommendations_Infra_demo/recommendations_demo.sh [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+current_dir="$(dirname "$0")"
+source ${current_dir}/../common_helper.sh
```
Comment (chandrams) [inline remote_monitoring_demo/recommendations_Infra_demo/recommendations_demo.sh:19]: Path of common_helper.sh needs to be corrected

File: remote_monitoring_demo/recommendations_Infra_demo/recommendations_demo.sh [hunk 5]
```diff
@@ -0,0 +1,440 @@
...
+current_dir="$(dirname "$0")"
+source ${current_dir}/../../common_helper.sh
+source ${current_dir}/recommendations_demo/recommendation_helper.sh
+# Default docker image repos
+AUTOTUNE_DOCKER_REPO="docker.io/kruize/autotune_operator"
+
+echo "${current_dir}/recommendations_demo/recommendation_helper.sh"
+
+# Default cluster
+CLUSTER_TYPE="openshift"
```
Comment (chandrams) [inline remote_monitoring_demo/recommendations_Infra_demo/recommendations_demo.sh:27]: @kusumachalasani - Running this on openshift fails with the below error, please fix this.

```
./recommendations_demo.sh -c openshift  -r --validate
```

ERROR: Please install minikube and try again
       Minikube resource config needed for demo:
       CPUs=8, Memory=16384MB

Comment (kusumachalasani) [reply remote_monitoring_demo/recommendations_Infra_demo/recommendations_demo.sh]: Done.

File: recommendations_Infra_demo/recommendations_demo/kruize/json_validate.py (hunk 3 above)
Comment (kusumachalasani) [reply recommendations_Infra_demo/recommendations_demo/kruize/json_validate.py]: Will take it up as a different PR as the same needs to changed in remote_monitoring tests  as well.

**Review by dinogun — APPROVED** [review]
LGTM

**Pull request merged**